File: algo/problems/0217-contains-duplicate/go/containsduplicate.go
package containsduplicate

// ContainsDuplicate 用 set（map[int]struct{}）边遍历边查。O(n)/O(n)。
func ContainsDuplicate(nums []int) bool {
	seen := make(map[int]struct{}, len(nums))
	for _, x := range nums {
		if _, ok := seen[x]; ok {
			return true
		}
		seen[x] = struct{}{}
	}
	return false
}

File: algo/problems/0217-contains-duplicate/java/ContainsDuplicate.java
import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {
    // HashSet.add 返回 false 即表示已存在 -> 命中重复。O(n)/O(n)。
    public static boolean containsDuplicate(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        for (int x : nums) {
            if (!seen.add(x)) {
                return true;
            }
        }
        return false;
    }
}
